src/tests/all.rs

```diff
@@ -855,9 +855,14 @@ impl MockUserSession {
     //     unimplemented!();
     // }
 
-    /// Obtain the database connection
-    pub fn db_conn(&self) -> DieselConnection {
-        self.app.diesel_database.get().unwrap()
+    /// Obtain the database connection and pass it to the closure
+    ///
+    /// Within each test, the connection pool only has 1 connection so it is necessary to drop the
+    /// connection before making any API calls.  Once the closure returns, the connection is
+    /// dropped, ensuring it is available for any API calls.
```

Comment:
This is nice! I think multiple people have hit this problem when starting to write new tests, and this API is much better than having to add curly braces for what might seem like a magical reason.